app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,94 @@
+class TweetController < ApplicationController
+
+	get '/tweets' do 
+		if logged_in?
+			@tweets = Tweet.all
+			erb :'/tweets/tweets'
+		else
+			redirect "/login"
+		end
+	end
+
+	get '/tweets/new' do 
+		if logged_in?
+
+			if !session[:content] 
+				session[:content] = true
+				@content = false
+			end
+
+			erb :'/tweets/create_tweet'
+		else
+			redirect to "/login"
+		end
+	end
+
+	post '/tweets' do 
+		content = Tweet.new(content: params[:content])
+
+		if !!content.save
+			current_user.tweets << content
+			current_user.save
+		else 
+		session[:content] = false
+		redirect '/tweets/new'	
+		end
+	end
+
+	get '/tweets/:id' do 
+		if logged_in?
+			@tweet = Tweet.find(params[:id])
+
+			@current_user = current_user
+			if session[:tweet] == "delete"
+				session[:tweet] = ""
+				@delete = "delete"
+			elsif session[:tweet] == "edit"
+				session[:tweet] = ""
+				@edit = "edit"
+			end
+
+			erb :'/tweets/show_tweet'
+		else
+			redirect "/login"
+		end	
+	end
+
+	delete '/tweets/:id/delete' do 
+		tweet = Tweet.find(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.